python/podio_class_generator.py

```diff
@@ -1034,17 +1033,22 @@ def prepare_vectorized_access(self, classname,members ):
         declaration += "\ttemplate<size_t arraysize>\n\tconst std::array<%s,arraysize> %s() const;\n" %(klass, name)
       return declaration, implementation
 
-    def write_file(self, name,content):
+    def write_file(self, name, content):
       #dispatch headers to header dir, the rest to /src
       # fullname = os.path.join(self.install_dir,self.package_name,name)
       if name.endswith("h"):
         fullname = os.path.join(self.install_dir,self.package_name,name)
       else:
         fullname = os.path.join(self.install_dir,"src",name)
       if not self.dryrun:
-        open(fullname, "w").write(content)
         if self.clang_format:
-          subprocess.call(self.clang_format + [fullname])
+          cfproc = subprocess.Popen(self.clang_format, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+          content = cfproc.communicate(input=content)[0]
+        existing_content = ''
+        if os.path.exists(fullname):
+          existing_content = open(fullname, 'r').read()
+        if existing_content != content:
+          open(fullname, 'w').write(content)
```

Comment:
These lines could be slightly more "pythonic" with a `with` statement: 
```python
with open(fullname, "r+") as _f:
  existing_content = _f.read()
  if existing_content != content:
    _f.write(content)
```

Reply:
```python
#!/usr/bin/env python
content ='foo bar'
fullname = 'temp.txt'
with open(fullname, "r+") as _f:
  existing_content = _f.read()
  if existing_content != content:
    _f.write(content)
```
Errors if temp.txt does not exist
```
$ python testRead.py
Traceback (most recent call last):
  File "testRead.py", line 6, in <module>
    with open(fullname, "r+") as _f:
IOError: [Errno 2] No such file or directory: 'temp.txt'
```
So we would still need one path for exists, one without.

Reply:
That's true, I overlooked that in one case the file won't exist yet. Taking that in consideration I think the code is best as it is!

Reply:
There are many things to be cleaned up, especially in view of python3 compatibility (#66)